// Copyright 2022 PingCAP, Ltd.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.#pragma once

syntax = "proto3";

package DB.DM.Remote;

service S3Lock {
    rpc tryAddLock(TryAddLockRequest) returns (TryAddLockResponse) {}
    rpc tryMarkDelete(TryMarkDeleteRequest) returns (TryMarkDeleteResponse) {}
}

message TryAddLockRequest {
    string ori_data_file = 1;
    uint64 ori_store_id = 2;
    uint64 lock_store_id = 3;
    uint64 upload_seq = 4;
}

message TryAddLockResponse {
    bool is_success = 1;
}

message TryMarkDeleteRequest {
    string ori_data_file = 1;
    uint64 ori_store_id = 2;
}

message TryMarkDeleteResponse {
    bool is_success = 1;
}
